xcm/xcm-builder/src/integration_tests.rs

```diff
@@ -0,0 +1,282 @@
+		];
+		let weight = 3 * BaseXcmWeight::get();
+		let r = XcmExecutor::<XcmConfig>::execute_xcm(
+			Parachain(PARA_ID).into(),
+			Xcm::WithdrawAsset {
+				assets: vec![ConcreteFungible { id: Null, amount }],
+				effects: vec![
+					buy_execution(weight),
+					Order::InitiateTeleport {
+						assets: vec![All],
+						dest: Parachain(other_para_id).into(),
+						effects: teleport_effects.clone(),
+					},
+				],
+			},
+			weight,
+		);
+		// teleports not allowed to community chains
+		assert_eq!(r, Outcome::Incomplete(weight, XcmError::UntrustedTeleportLocation));
+		let r = XcmExecutor::<XcmConfig>::execute_xcm(
+			Parachain(PARA_ID).into(),
+			Xcm::WithdrawAsset {
+				assets: vec![ConcreteFungible { id: Null, amount }],
+				effects: vec![
+					buy_execution(weight),
+					Order::InitiateTeleport {
+						assets: vec![All],
+						dest: Parachain(statemine_id).into(),
+						effects: teleport_effects.clone(),
+					},
+				],
+			},
+			weight,
+		);
+		assert_eq!(r, Outcome::Complete(weight));
+		// 2 * amount because of the incomplete failed attempt above
+		assert_eq!(Balances::free_balance(para_acc), INITIAL_BALANCE - 2 * amount);
+		assert_eq!(
+			mock::sent_xcm(),
+			vec![(
+				Parachain(statemine_id).into(),
+				Xcm::TeleportAsset {
+					assets: vec![ConcreteFungible { id: Parent.into(), amount }],
+					effects: teleport_effects,
+				}
+			)]
+		);
+	});
+}
+
+/// Scenario:
+/// A parachain wants to move KSM from Kusama to the parachain.
+/// It withdraws funds and then deposits them into the reserve account of the destination chain.
+/// to the destination.
+///
+/// Asserts that the balances are updated accordingly and the correct XCM is sent.
+#[test]
+fn reserve_based_transfer_works() {
+	use xcm::opaque::v0::prelude::*;
+	let para_acc: AccountId = ParaId::from(PARA_ID).into_account();
+	let balances = vec![(ALICE, INITIAL_BALANCE), (para_acc.clone(), INITIAL_BALANCE)];
+	kusama_like_with_balances(balances).execute_with(|| {
+		let other_para_id = 3000;
+		let amount =  10 * ExistentialDeposit::get();
+		let transfer_effects = vec![
+			buy_execution(5), // unchecked mock value
+			Order::DepositAsset { assets: vec![All], dest: X2(Parent, Parachain(PARA_ID)) },
+		];
+		let weight = 3 * BaseXcmWeight::get();
+		let r = XcmExecutor::<XcmConfig>::execute_xcm(
+			Parachain(PARA_ID).into(),
+			Xcm::WithdrawAsset {
+				assets: vec![ConcreteFungible { id: Null, amount }],
+				effects: vec![
+					buy_execution(weight),
+					Order::DepositReserveAsset {
+						assets: vec![All],
+						dest: Parachain(other_para_id).into(),
+						effects: transfer_effects.clone(),
+					},
+				],
+			},
+			weight,
+		);
+		assert_eq!(r, Outcome::Complete(weight));
+		assert_eq!(Balances::free_balance(para_acc), INITIAL_BALANCE - amount);
+		assert_eq!(
+			mock::sent_xcm(),
+			vec![(
+				Parachain(other_para_id).into(),
+				Xcm::ReserveAssetDeposit {
```

Comment:
```suggestion
				Xcm::ReserveAssetDeposited {
```

didn't we decide to chnage this? :D

Reply:
yes, it's just not up-to-date